nomad/node_endpoint.go

```diff
@@ -154,7 +154,8 @@ func (n *Node) UpdateStatus(args *structs.NodeUpdateStatusRequest, reply *struct
 	}
 
 	// Check if we should trigger evaluations
-	if structs.ShouldDrainNode(args.Status) {
+	initToReady := node.Status == structs.NodeStatusInit && args.Status == structs.NodeStatusReady
```

Comment:
It should also trigger if we go from a Terminal state to Ready. e.g. Node is ready, then goes down (system scheduler stops the allocs), node comes back and goes ready (system scheduler must create new allocs)